lib/src/form_builder_validators.dart

```diff
@@ -307,11 +356,240 @@ class FormBuilderValidators {
               : null;
 
   /// [FormFieldValidator] that requires the field's value to be a valid date string.
-  static FormFieldValidator<String> dateString({
+  static FormFieldValidator<String> date({
     String? errorText,
   }) =>
       (valueCandidate) => true == valueCandidate?.isNotEmpty &&
               !isDate(valueCandidate!)
           ? errorText ?? FormBuilderLocalizations.current.dateStringErrorText
           : null;
+
+  /// [FormFieldValidator] that requires the field's value to be a date within a certain range.
+  static FormFieldValidator<String> dateRange({
+    required String minDate,
+    required String maxDate,
+    String? errorText,
+  }) {
+    return compose<String>(
+      [
+        minDate.isNotEmpty
+            ? date(errorText: errorText)
+            : (valueCandidate) => null,
+        maxDate.isNotEmpty
+            ? date(errorText: errorText)
+            : (valueCandidate) => null,
+        (valueCandidate) {
+          if (valueCandidate == null || valueCandidate.isEmpty) {
+            return null;
+          }
+
+          final minDateTime = DateTime.tryParse(minDate);
+          final maxDateTime = DateTime.tryParse(maxDate);
+          final valueDateTime = DateTime.tryParse(valueCandidate);
+
+          if (minDateTime != null &&
+              valueDateTime!.isBefore(minDateTime) &&
+              maxDateTime != null &&
+              valueDateTime.isAfter(maxDateTime)) {
+            return errorText ??
+                FormBuilderLocalizations.current.dateRangeErrorText(
+                  minDate,
+                  maxDate,
+                );
+          }
+
+          return null;
+        },
+      ],
+    );
+  }
+
+  /// [FormFieldValidator] that requires the field's value to be a valid phone number.
+  static FormFieldValidator<String> phoneNumber({
+    String? errorText,
+  }) =>
+      (valueCandidate) =>
+          true == valueCandidate?.isNotEmpty && !isPhoneNumber(valueCandidate!)
+              ? errorText ?? FormBuilderLocalizations.current.phoneErrorText
+              : null;
```

Comment:
I shouldn't this validator. Is really hard validate a telephone number for every localization. Exist a package only for this: https://github.com/flutter-form-builder-ecosystem/form_builder_phone_field

Reply:
Yeah that's why it is not so strict in validating right now. It's more a basic sanity check. I think it is ok to add like this and others could make a PR if it doesn't work well enough.

Reply:
I see no sense in adding a validator that does not offer the functionality that the name indicates.
What is the use cases that this validation is useful? Only US numbers?

Reply:
I've added a bunch of tests to cover most cases.